eng/finalize-publish.yml

```diff
@@ -0,0 +1,74 @@
+jobs:
+  - job: Finalize_Publish_Versions
+    displayName: Finalize_Publish_Versions
+    dependsOn:
+    - build_Linux_arm_release
+    - build_Linux_arm64_release
+    - build_Linux_musl_x64_release
+    - build_Linux_rhel6_x64_release
+    - build_Linux_x64_release
+    - build_FreeBSD_x64_release
```

Comment:
You should depend on the publish-build-assets leg instead of each individual leg for a few reasons:
* Avoids duplicating the info. (It already doesn't match: this FreeBSD leg isn't in the other copy)
* Ensures the build actually publishes before posting about it on dotnet/versions. If the publish broke or takes a long time, we can't put info on dotnet/versions yet.

Reply:
Sure.
It looked like core-setup didn't follow this pattern, but maybe I don't understand arcade yaml enough yet.

Reply:
In Core-Setup, finalize-publish is the one publish leg, so there is no duplication. Here there's the existing publish leg, plus this new leg being added in the PR. Core-Setup doesn't use the Arcade yml yet except to publish to BAR.